src/Swift.Bindings/src/Emitter/StringEmitter/Handler/MethodHandler.cs

```diff
@@ -385,22 +385,38 @@ internal static class PInvokeEmitter
         /// <summary>
         /// Emits the PInvoke signature.
         /// </summary>
-        /// <param name="writer">The IndentedTextWriter instance.</param>
+        /// <param name="csWriter">The IndentedTextWriter instance.</param>
         /// <param name="methodEnv">The method environment.</param>
-        public static void EmitPInvoke(IndentedTextWriter writer, MethodEnvironment methodEnv, SignatureHandler signatureHandler)
+        public static void EmitPInvoke(CSharpWriter csWriter, MethodEnvironment methodEnv, SignatureHandler signatureHandler)
         {
             var methodDecl = (MethodDecl)methodEnv.MethodDecl;
             var moduleDecl = methodDecl.ModuleDecl ?? throw new ArgumentNullException(nameof(methodDecl.ModuleDecl));
 
             var pInvokeName = NameProvider.GetPInvokeName(methodDecl);
             var libPath = methodEnv.TypeDatabase.GetLibraryPath(moduleDecl.Name);
 
-            writer.WriteLine("[UnmanagedCallConv(CallConvs = new Type[] { typeof(CallConvSwift) })]");
-            writer.WriteLine($"[DllImport(\"{libPath}\", EntryPoint = \"{methodDecl.MangledName}\")]");
-
             var pInvokeSignature = signatureHandler.GetPInvokeSignature();
 
-            writer.WriteLine($"private static extern {pInvokeSignature.ReturnType} {pInvokeName}({pInvokeSignature.ParametersString()});");
+            csWriter.WriteLine("[UnmanagedCallConv(CallConvs = new Type[] { typeof(CallConvSwift) })]");
+            if (methodDecl.IsAsync)
+            {
+                string parameters = string.Join(", ",
```

Comment:
This can go into SignatureBuilder. EntryPoint could go into NameProvider, this way this function wouldn't even need to have an if statement

Reply:
Added specific handler in the PInvokeSignature.